src/MongoDB.Driver/IAggregateFluent.cs

```diff
@@ -206,6 +206,27 @@ public interface IAggregateFluent<TResult> : IAsyncCursorSource<TResult>
         /// <returns>The fluent aggregate interface.</returns>
         IAggregateFluent<TNewResult> Lookup<TForeignDocument, TNewResult>(string foreignCollectionName, FieldDefinition<TResult> localField, FieldDefinition<TForeignDocument> foreignField, FieldDefinition<TNewResult> @as, AggregateLookupOptions<TForeignDocument, TNewResult> options = null);
 
+        /// <summary>
+        /// Appends a lookup stage to the pipeline.
+        /// </summary>
+        /// <typeparam name="TForeignDocument">The type of the foreign document.</typeparam>
+        /// <typeparam name="TAsElement">The inner type of <typeparamref name="TAs" /> collection.</typeparam>
+        /// <typeparam name="TAs">The type of <typeparamref name="TAs" /> collection.</typeparam>
+        /// <typeparam name="TNewResult">The type of the new result.</typeparam>
+        /// <param name="foreignCollectionName">Name of the other collection.</param>
+        /// <param name="let">The "let" definition.</param>
+        /// <param name="pipeline">The pipeline.</param>
+        /// <param name="as">The field in <typeparamref name="TNewResult" /> to place the foreign results.</param>
+        /// <param name="options">The options.</param>
+        /// <returns>The fluent aggregate interface.</returns>
+        IAggregateFluent<TNewResult> Lookup<TForeignDocument, TAsElement, TAs, TNewResult>(
+            string foreignCollectionName,
+            BsonDocument let,
+            PipelineDefinition<TForeignDocument, TAsElement> pipeline,
+            FieldDefinition<TNewResult, TAs> @as,
+            AggregateLookupOptions<TForeignDocument, TNewResult> options = null)
+            where TAs : IEnumerable<TAsElement>;
+
```

Comment:
I think we need a new Lookup extension method (or two?) in IAggregateFluentExtensions.cs that takes an Expression for the @as parameter.